port/unix/omrsysinfo.c

```diff
@@ -1564,14 +1583,25 @@ omrsysinfo_get_s390_description(struct OMRPortLibrary *portLibrary, OMRProcessor
 	/* zEC12 facility and processor detection */
 
 	/* TE/TX hardware support */
-	if (omrsysinfo_test_stfle(portLibrary, 50) && omrsysinfo_test_stfle(portLibrary, 73)) {
+	if (omrsysinfo_test_stfle(portLibrary, 73)) {
 #if defined(J9ZOS390)
 		if (omrsysinfo_get_s390_zos_supports_transactional_execution_facility())
 #elif defined(LINUX) /* LINUX S390 */
 		if (OMR_ARE_ALL_BITS_SET(auxvFeatures, OMR_HWCAP_S390_TE))
 #endif /* defined(J9ZOS390) */
 		{
-			omrsysinfo_set_feature(desc, OMR_FEATURE_S390_TE);
+			omrsysinfo_set_feature(desc, OMR_FEATURE_S390_TRANSACTIONAL_EXECUTION_FACILITY);
+		}
+	}
+
+	if (omrsysinfo_test_stfle(portLibrary, 50)) {
```

Comment:
From Principles of Operation, the bit 50 of STFLE is only valid if bit 73 is 1.  As such, I think we should fold this constrained transaction check into the above if-condition for 73.  I worry that some hypervisor configuration may get us into a mode where 50 is set, but 73 is not.

>50 The constrained transactional-execution facility is installed in the z/Architecture architectural mode. **This bit is meaningful only when bit 73 is one.**